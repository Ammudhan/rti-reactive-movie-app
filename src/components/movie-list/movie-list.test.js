import React from "react";
import renderer, { create } from "react-test-renderer";
import MovieList from "./movie-list";
import MovieService from "../../services/movieService";
import {StaticRouter} from "react-router-dom";

describe("Movie List Component", () => {

    it("should define the state as expected", async () => {
        const mockData = {
           data: {
                results: [
                    {
                        id: 429617,
                        original_title: "Spider-Man: Far from Home",
                        poster_path: "/rjbNpRMoVvqHmhmksbokcyCr7wn.jpg"
                    }
                ]
            }
        };
        const mock = jest.spyOn(MovieService, "getMovies");
        const promise = Promise.resolve(mockData);
        mock.mockImplementation(() => promise);
        create(
            <StaticRouter>
                <MovieList/>
            </StaticRouter>
        );
        await promise;
        expect(mock).toHaveBeenCalled();
    });

    it('should matches snapshot', async () => {
        const mockData = {
            data: {
                results: [
                    {
                        id: 429617,
                        original_title: "Spider-Man: Far from Home",
                        poster_path: "/rjbNpRMoVvqHmhmksbokcyCr7wn.jpg"
                    }
                ]
            }
        };
        const mock = jest.spyOn(MovieService, "getMovies");
        const promise = Promise.resolve(mockData);
        mock.mockImplementation(() => promise);
        const component = renderer.create(
            <StaticRouter>
                <MovieList/>
            </StaticRouter>
        );
        await promise;
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});