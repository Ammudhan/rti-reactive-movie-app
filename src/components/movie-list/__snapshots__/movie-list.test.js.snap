// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List Component should matches snapshot 1`] = `
<span
  className="row"
>
  <span
    className="col-6 col-md-4 col-lg-3 rti-no-gutter"
  >
    <a
      href="/movie"
      onClick={[Function]}
    >
      <img
        alt="Spider-Man: Far from Home"
        className="img-fluid link"
        src="https://image.tmdb.org/t/p/original/rjbNpRMoVvqHmhmksbokcyCr7wn.jpg"
      />
    </a>
  </span>
</span>
`;
